lib/service/Service.ts

```diff
@@ -193,6 +196,9 @@ class Service extends EventEmitter {
    * Connect to an XU node on a given host and port.
    */
   public connect = async ({ host, port }: { host: string, port: number }) => {
+    checkArgument(host !== '', 'host must be specified');
+    checkArgument(port > 1023 && port < 65536, 'port must be an integer between 1024 and 65535');
```

Comment:
I think we should wrap `checkArgument` calls we use multiple times into a function or at least define the messages of those calls in one place so that they stay consistent.